### ISSUE TITLE
Document insertion with persistent index over array fails

### ISSUE THREAD

**Issue by miguel-porto** [arangodb/arangodb#14546]
## My Environment

* __ArangoDB Version__:  3.7.13
* __Storage Engine__: RocksDB
* __Deployment Mode__:  Single Server
* __Deployment Strategy__:  Manual Start
* __Configuration__: 
* __Infrastructure__:   own
* __Operating System__: Ubuntu 20.04
* __Total RAM in your machine__: 12Gb
* __Disks in use__:  SSD
* __Used Package__:  Ubuntu .deb

## Component, Query & Data

__Affected feature__:
Insert document with persistent index in array

__AQL query (if applicable)__:
Sequence of 3 queries:
```
INSERT {"occurrences":[{"uuid":"abcd"}]} INTO inventory

// remove value from 'occurrences' array:
FOR i IN inventory
    LET newo = (FOR o IN i.occurrences
        FILTER o.uuid != 'abcd'
        RETURN o)
    UPDATE i WITH {occurrences: newo} IN inventory

// insert document with same uuid
INSERT {"occurrences":[{"uuid":"abcd"}]} INTO inventory
// Fails!
```

__Dataset__:
I can send the debug package if needed, but it's irrelevant, it is easy to reproduce with a new database (see below)

__Size of your Dataset on disk__:
?

## Steps to reproduce
1. Create a database with one collection "inventory"
2. Create a persistent index over "occurrences[*].uuid", unique and sparse
3. Execute the three queries in sequence

__Problem__:
The re-insertion of the document with the same uuid that was used before, fails with error, although it has been removed
com.arangodb.ArangoDBException: Response: 404, Error: 1202 - NotFound:  - in index uuid of type hash over 'occurrences[*].uuid'

__Expected result__:
The document should be inserted because the uuid value is no longer in use. It looks like the uuid value is not removed from the index, despite that it has been removed from the document.

**Comment by Simran-B** [comment]
I was able to reproduce the problem and created an internal ticket BTS-548. Thank you @miguel-porto for reporting this!

**Comment by Simran-B** [comment]
The problem is caused by an incorrect optimization that prevents the removal of the index entry. Will be fixed in:

- 3.7.15 https://github.com/arangodb/arangodb/pull/14781
- 3.8.2 https://github.com/arangodb/arangodb/pull/14780
- devel https://github.com/arangodb/arangodb/pull/14779

**Comment by miguel-porto** [comment]
Thank you @Simran-B for covering this issue! The description of the problem in the PR is unequivocal, that's it.

**Comment by miguel-porto** [comment]
I confirm this is fixed in 3.7.15! Thank you very much!
